onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -1,5 +1,10 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4244)
+#endif
+
```

Comment:
Why is it necessary to disable this vs fixing the warning?

Reply:
This warning seems to be disabled at [multiple places](https://github.com/microsoft/onnxruntime/search?p=1&q=%23pragma+warning%28disable+%3A+4244%29&unscoped_q=%23pragma+warning%28disable+%3A+4244%29) in the codebase from master, for example see [graph.cc line 4-7.](https://github.com/microsoft/onnxruntime/blob/95ade8f47bc257a152bab53dc44d9aae0b861cdd/onnxruntime/core/graph/graph.cc#L6). 

@skottmckay , does it make sense to single out ort_training?

Reply:
Spoke offline, @skottmckay will not block on this. There is a task open to investigate this warning that is spread across the codebase, we will add a child item to it for this file. You may assign that child item to me.

CC: @edgchen1